test/unit/org/apache/cassandra/io/sstable/format/bti/RowIndexTest.java

```diff
@@ -0,0 +1,517 @@
+                    assertTrue(comparator.compare(left, keys.get(idx + 1)) < 0);
+                // Because of the way we build the index (using non-prefix separator) we are usually going to miss the last item.
+                if (idx >= 2)
+                    assertTrue(comparator.compare(left, keys.get(idx - 2)) > 0);
+            }
+            catch (AssertionError e)
+            {
+                logger.error(e.getMessage(), e);
+                ClusteringPrefix<?> ll = left;
+                ClusteringPrefix<?> rr = right;
+                logger.info(keys.stream()
+                                .filter(x -> comparator.compare(ll, x) <= 0 && comparator.compare(x, rr) <= 0)
+                                .map(clustering -> comparator.asByteComparable(clustering))
+                                .map(bc -> bc.byteComparableAsString(VERSION))
+                                .collect(Collectors.joining(", ")));
+                logger.info("Left {}{} Right {}{}", comparator.asByteComparable(left), exactLeft ? "#" : "", comparator.asByteComparable(right), exactRight ? "#" : "");
+                try (RowIndexReverseIterator iter2 = new RowIndexReverseIterator(fh, root, comparator.asByteComparable(left), comparator.asByteComparable(right)))
+                {
+                    IndexInfo ii;
+                    while ((ii = iter2.nextIndexInfo()) != null)
+                    {
+                        logger.info(comparator.asByteComparable(keys.get((int) ii.offset)).toString());
+                    }
+                    logger.info("Left {}{} Right {}{}", comparator.asByteComparable(left), exactLeft ? "#" : "", comparator.asByteComparable(right), exactRight ? "#" : "");
+                }
+                throw e;
+            }
+        }
+    }
+
+    @Test
+    public void testReverseIteration() throws IOException
+    {
+        Pair<List<ClusteringPrefix<?>>, RowIndexReader> random = generateRandomIndexSingletons(COUNT);
+        List<ClusteringPrefix<?>> keys = random.left;
+
+        for (int i = 0; i < 1000; ++i)
+        {
+            boolean exactRight = RANDOM.nextBoolean();
+            ClusteringPrefix<?> right = exactRight ? keys.get(RANDOM.nextInt(keys.size())) : generateRandomKey();
+
+            int idx = 0;
+            try (RowIndexReverseIterator iter = new RowIndexReverseIterator(fh, root, ByteComparable.EMPTY, comparator.asByteComparable(right)))
+            {
+                IndexInfo indexInfo = iter.nextIndexInfo();
+                if (indexInfo == null)
+                {
+                    idx = Collections.binarySearch(keys, right, comparator);
+                    if (idx < 0)
+                        idx = -2 - idx; // less than or equal
+                    assertTrue(comparator.asByteComparable(keys.get(idx)) + " <= "
+                               + comparator.asByteComparable(right) + " but " + idx + " wasn't iterated.",
+                               idx < 0);
```

Comment:
Static analysis is telling me this is always `false`, but it seems like idx could still be `-1`?